synapse/storage/databases/main/cache.py

```diff
@@ -59,6 +60,7 @@ class CacheInvalidationWorkerStore(
     RelationsWorkerStore,
     EventPushActionsWorkerStore,
     StreamWorkerStore,
+    StatsStore,
```

Comment:
Instead of this we could use `_clock`, which is provided by `SQLBaseStore`. That might be a separate PR though?

Reply:
I like the sound of that. I really didn't like making `CacheInvalidationWorkerStore` depend on stats!

Reply:
I still need something to provide `self._curr_state_delta_stream_cache` though, so I'll have to inherit from `StateDeltasStore` for now.

Reply:
I agree! While we're at it, maybe make `stats` just use the one from `SQLBaseStore` too?

Reply:
In the future I think we could refactor the (only) usage of _curr_state_delta_stream_cache to call a method on StateDeltasStore.

Reply:
Sure, I meant specifically the `clock` just to be clear!

Reply:
Ahh, I think our comments raced each other! I was thinking of a way to make `CacheInvalidationWorkerStore` less icky.

But agreed, I'm only planning to adjust `StatsStore.clock`.